test/tests/functional/pbs_power_provisioning_cray.py

```diff
@@ -449,15 +500,536 @@ def test_last_used_time(self):
+
+    @timeout(1500)
+    def test_power_on_min_node_down_delay(self):
+        """
+        Test when a job is calandered on a vnode which is in sleep state,
+        the node will be not be powered on until min_node_down_delay time.
+        """
+        self.scheduler.set_sched_config({'strict_ordering': 'True ALL'})
+        self.server.manager(MGR_CMD_SET, NODE, {
+                            "poweroff_eligible": True}, self.names[0])
+        a = {"power_on_off_enable": True, 'min_node_down_delay': '3000'}
+        self.modify_hook_config(attrs=a, id='PBS_power')
+        a = {'freq': 30}
+        self.server.manager(MGR_CMD_SET, PBS_HOOK, a, id='PBS_power',
+                            sudo=True)
+        a = {'enabled': 'True'}
+        self.server.manager(MGR_CMD_SET, PBS_HOOK, a, id='PBS_power',
+                            sudo=True)
+        self.logger.info("Waiting for 1 min to poweroff 1st node")
+        time.sleep(60)
+        self.server.expect(NODE, {'state': 'sleep'}, id=self.names[0])
+        jid = self.submit_job(1000, {'Resource_List.vnode': self.names[0],
+                                     'Resource_List.vntype': 'cray_compute'})
+        self.server.manager(MGR_CMD_SET, SERVER, {'scheduling': 'True'})
+        self.logger.info("Waiting for 2 mins to make sure node is not powered")
+        time.sleep(120)
+        self.server.expect(JOB, {'job_state': 'Q'}, id=jid)
+        self.server.expect(NODE, {'state': 'sleep'}, id=self.names[0])
+        self.cleanup_power_on()
+
+    @timeout(1500)
+    def test_max_jobs_analyze_limit(self):
+        """
+        Test that even when 4 jobs are calandered only nodes assigned
+        to max_jobs_analyze_limit number of jobs will be considered
+        for powering on.
+        """
+        self.scheduler.set_sched_config({'strict_ordering': 'True ALL'})
+        self.server.manager(MGR_CMD_SET, SERVER, {'backfill_depth': '4'})
+        for n in self.server.status(NODE):
+            if 'resources_available.PBScraynid' in n:
+                self.server.manager(MGR_CMD_SET, NODE, {
+                                    "poweroff_eligible": True}, n['id'])
+        a = {"power_on_off_enable": True, 'max_jobs_analyze_limit': '2',
+             'node_idle_limit': '30', 'max_concurrent_nodes': '30'}
+        self.modify_hook_config(attrs=a, id='PBS_power')
+        a = {'freq': 30}
+        self.server.manager(MGR_CMD_SET, PBS_HOOK, a, id='PBS_power',
+                            sudo=True)
+        a = {'enabled': 'True'}
+        self.server.manager(MGR_CMD_SET, PBS_HOOK, a, id='PBS_power',
+                            sudo=True)
+        self.server.manager(MGR_CMD_SET, SERVER, {'backfill_depth': '10'})
+        self.logger.info("Waiting for 2 mins to poweroff all the nodes")
+        time.sleep(120)
+        for n in self.server.status(NODE):
+            if 'resources_available.PBScraynid' in n:
+                self.server.expect(NODE, {'state': 'sleep'}, id=n['id'])
```

Comment:
As above, use the list compression or self.server.expect(NODE, {'state=sleep': len(self.server.status(NODE)))